rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -141,14 +152,40 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   }
 
   template<typename T>
-  typename std::enable_if<std::is_same<T, rcl_serialized_message_t>::value, void>::type
+  typename std::enable_if<
+    std::is_same<T, rcl_serialized_message_t>::value &&
+    !std::is_same<MessageT, rclcpp::experimental::SerializedContainer>::value,
+    void>::type
   execute_impl()
   {
-    throw std::runtime_error("Subscription intra-process can't handle serialized messages");
+    if (serializer_) {
```

Comment:
nitpick here: I would invert the check and throw/return early `if (nullptr == serializer_)`